--- old/boost/detail/endian.hpp	2012-06-14 01:30:13.259839531 +0200
+++ boost/boost/detail/endian.hpp	2012-06-14 01:37:57.459818832 +0200
@@ -31,7 +31,7 @@
 // GNU libc offers the helpful header <endian.h> which defines
 // __BYTE_ORDER
 
-#if defined (__GLIBC__)
+#if defined (__GLIBC__) || defined(ANDROID)
 # include <endian.h>
 # if (__BYTE_ORDER == __LITTLE_ENDIAN)
 #  define BOOST_LITTLE_ENDIAN
